api/src/main/java/io/opentelemetry/common/AttributeValue.java

```diff
@@ -244,6 +244,15 @@ public double getDoubleValue() {
    */
   public abstract Type getType();
 
+  /**
+   * Returns {@code true} if the {@code AttributeValue} contains a {@code null} value or it is an
+   * empty array.
+   *
+   * @return {@code true} if the {@code AttributeValue} is empty.
+   * @since 0.8.0
+   */
+  public abstract boolean isEmpty();
```

Comment:
we could just have this be concrete and return `false` and only override it for the types that support emptiness.